File: lib/day01/solution.go
package main

import (
	"fmt"
	"sort"
	"strconv"
	"strings"

	"adventofcode.com/lib/commons"
)

func main() {
	text, _ := commons.ReadFileAsText("lib/day01/input.txt")

	var sum []int
	var tempList []int

	stringList := strings.Split(text, "\n")

	for _, item := range stringList {
		if item == "" {
			tempSum := 0
			for _, num := range tempList {
				tempSum += num
			}
			sum = append(sum, tempSum)
			tempList = nil
		} else {
			num, err := strconv.Atoi(item)
			if err != nil {
				panic(err)
			}
			tempList = append(tempList, num)
		}
	}

	sort.Slice(sum, func(i, j int) bool {
		return sum[i] < sum[j]
	})
	max := sum[len(sum)-1]
	fmt.Println("The maximum sum is:", max)

}

File: lib/day01/solution.dart
import 'package:advent_of_code_2022/commons/read_file_as.dart';

class Day01 {
  static int soltion01() {
    var text = readFileAsText('lib/day01/input.txt');
    List<int> sum = [];
    List<int> tempList = [];
    List<String> stringList = text.split('\n');
    for (var item in stringList) {
      if (item.isEmpty) {
        sum.add(tempList.reduce((a, b) => a + b));
        tempList = [];
      } else {
        tempList.add(int.parse(item));
      }
    }
    //get maximum sum
    sum.sort((a, b) => b.compareTo(a));
    print(sum.first);
    return sum.first;
  }
}
